### PR TITLE
359 years of life lost 5 yrs

### LINKED ISSUE #359: VZV: Number of years of life lost viz
monthly or yearly chart for years of life lost

### PR DESCRIPTION
**Pull request by tcwcodes**
Closes #359. This creates the more robust years of life lost YTD 5-year timeline I suggested. It also deletes the GraphQL query that we no longer need to be saved in the queries folder.

While this closes 359, that issue has valuable comments and conversations that are still ongoing. @mateoclarke  and @JaceDeloney, what would be the best way to preserve those comments/conversations - moving them to an epic?

Screen Shot 2019-11-21 at 6 24 31 PM
Screen Shot 2019-11-21 at 6 24 59 PM

### CODE REVIEW

File: atd-vzv/src/views/dashboard/yearsOfLifeLost.js [hunk 1]
```diff
@@ -11,15 +12,41 @@ const YearsOfLifeLost = () => {
...
+    .format("YYYY");
+  const fourYearsAgo = moment()
+    .subtract(4, "year")
+    .format("YYYY");
 
   const yearToDateUrl = `https://data.austintexas.gov/resource/xecs-rpy9.json?$where=prsn_injry_sev_id = '4' AND crash_date between '${thisYear}-01-01T00:00:00' and '${today}T23:59:59'`;
   const previousYearUrl = `https://data.austintexas.gov/resource/xecs-rpy9.json?$where=prsn_injry_sev_id = '4' AND crash_date between '${lastYear}-01-01T00:00:00' and '${lastYear}${todayMonthYear}T23:59:59'`;
+  const twoYearsAgoUrl = `https://data.austintexas.gov/resource/xecs-rpy9.json?$where=prsn_injry_sev_id = '4' AND crash_date between '${twoYearsAgo}-01-01T00:00:00' and '${twoYearsAgo}${todayMonthYear}T23:59:59'`;
+  const threeYearsAgoUrl = `https://data.austintexas.gov/resource/xecs-rpy9.json?$where=prsn_injry_sev_id = '4' AND crash_date between '${threeYearsAgo}-01-01T00:00:00' and '${threeYearsAgo}${todayMonthYear}T23:59:59'`;
+  const fourYearsAgoUrl = `https://data.austintexas.gov/resource/xecs-rpy9.json?$where=prsn_injry_sev_id = '4' AND crash_date between '${fourYearsAgo}-01-01T00:00:00' and '${fourYearsAgo}${todayMonthYear}T23:59:59'`;
```
Comment (mateoclarke) [inline atd-vzv/src/views/dashboard/yearsOfLifeLost.js]: For lines 12-29, you could refactor this code to be less repetitive...

```js
const getFatailitiesByYearsAgoUrl = (yearsAgo) => {
   let yearsAgoDate = moment().subtract(yearsAgo, "year").format("YYYY");

   `https://data.austintexas.gov/resource/xecs-rpy9.json?$where=prsn_injry_sev_id = '4' AND crash_date between '${yearsAgoDate}-01-01T00:00:00' and '${yearsAgoDate}${todayMonthYear}T23:59:59'`;
}
```


Then below around line 75, for each of the axios calls, you could do something like...

```js
axios.get(getFatailitiesByYearsAgoUrl(2)).then(res => {
  setTwoYearsAgoYearToDateYearsLostTotal(getYearsOfLifeLost(res.data));
});
```

File: atd-vzv/src/views/dashboard/yearsOfLifeLost.js [hunk 2]
```diff
@@ -38,34 +65,111 @@ const YearsOfLifeLost = () => {
...
+      `${thisYear}`
+    ],
+    datasets: [
+      {
+        label: "Years of Life Lost Year to Date",
+        fill: false,
+        lineTension: 0.1,
+        backgroundColor: "rgba(192,10,10,0.4)", // Legend box
+        borderColor: "rgba(192,10,10,1)",
+        borderCapStyle: "butt",
```
Comment (mateoclarke) [inline atd-vzv/src/views/dashboard/yearsOfLifeLost.js]: 🍑

**Comment by mateoclarke** [comment]
On the question of keeping the from closing the active conversation, I think you should make a new issue specific to the CTM privacy questions and assign it to @johnclary.

**Comment by JaceDeloney** [comment]
Created #467 for the privacy questions.

**Comment by JaceDeloney** [comment]
@tcwcodes Can we prioritize #454 ? 

@mateoclarke I would like to better understand what @tcwcodes had in mind for this visualization. Cumulative years of life lost is a very powerful metric that our users are interested seeing, but I'm unsure of the value that a year-over-year comparison provides. We should chat on how to make this year-over-year comparison more useful.

I'm going to de-prioritize it for now.

**Comment by tcwcodes** [comment]
I'm going to put a pin in this and move on to #454

**Comment by mateoclarke** [comment]
It looks like this code was included in the #468 PR and got merged when that PR did. Not sure if that was the intent but we can delete this code if we decide we don't want to keep this viz widget

**Pull request merged**